redwood-lazylayout-compose/src/commonMain/kotlin/app/cash/redwood/lazylayout/compose/LazyListState.kt

```diff
@@ -46,4 +47,17 @@ public class LazyListState(
     firstVisibleItemIndex = index
     scrollToItemTriggeredId++
   }
+  public companion object {
+    /**
+     * The default [Saver] implementation for [LazyListState].
+     */
+    public val Saver: Saver<LazyListState, *> = Saver(
+      save = { it.firstVisibleItemIndex },
+      restore = {
+        LazyListState(
+          firstVisibleItemIndex = it,
+        )
+      }
+    )
+  }
```

Comment:
this is copy-paste from https://github.com/cashapp/redwood/pull/1494, and will be reverted once this PR goes from draft -> ready for review